syntax = "proto3";

package virtualfriends_proto;

// Next tag: 7
message VfRequest {
    reserved 1;

    oneof request {
        EchoRequest echo = 2;

        StreamReplyMessageRequest stream_reply_message = 6;
    }
}

// Next tag: 7
message VfResponse {
    CustomError error = 1;

    oneof response {
        EchoResponse echo = 2;

        StreamReplyMessageResponse stream_reply_message = 6;
    }
}

// Next tag: 2
message CustomError {
    string error_message = 1;
}

// Next tag: 2
message EchoRequest {
    string text = 1;
}

// Next tag: 2
message EchoResponse {
    string text = 1;
}

// Next tag: 3
enum VoiceGender {
    VoiceGender_Invalid = 0;

    VoiceGender_Male = 1;

    VoiceGender_Female = 2;
}

// Next tag: 7
message StreamReplyMessageRequest {
    string character_name = 1;
    
    repeated string json_messages = 2;

    oneof current_message {
        bytes wav = 3;

        string text = 6;
    }

    int64 session_id = 4;

    VoiceGender gender = 5; 
}

// Next tag: 9
message StreamReplyMessageResponse {
    string reply_message = 1;

    string action = 2;

    string sentiment = 3;

    bytes reply_wav = 4;

    // will be omitted except the first one.
    string transcribed_text = 5;

    int64 session_id = 6;

    int32 chunk_index = 7;

    bool is_stop = 8;
}
